### DISCUSSION TITLE
Clarify "zero nonce"

### DISCUSSION THREAD

**Discussion by paulmillr** [FiloSottile/age#227] [Spec feedback]
`encrypt[key](plaintext) is ChaCha20-Poly1305 from RFC 7539 with a zero nonce.`

Everyone keeps saying we should never reuse nonces.

Is age using zero nonces because the underlying messages never repeat? If so, the spec should clarify this part. Also, what is zero? Empty byte array, or a byte array with `[0x0]`?

**Comment by mdouchement** [comment]
This comment should explain:
https://github.com/FiloSottile/age/blob/e43cf8b4a2d571df8dcb30783a460acbf188adb4/internal/age/primitives.go#L24-L28

**Comment by paulmillr** [comment]
Yeah, just would be great to have this clarified in the spec.

**Comment by paulmillr** [comment]
What about multi-target attacks? https://blog.cr.yp.to/20151120-batchattacks.html

**Reply by FiloSottile** [reply]
ChaCha20Poly1305 keys are 256 bits, so they have plenty of multi-target margin even with a fixed nonce. A version of the spec that uses AES-128 would have to generate the nonce from the same source as the key.
